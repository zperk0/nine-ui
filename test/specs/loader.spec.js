import NLoader from '../../packages/n-loader';
import { mount } from '@vue/test-utils';

describe('NLoader', () => {
  it('renders correctly', () => {
    const wrapper = mount({
      render () {
        return <NLoader show={true}/>;
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('render size correctly', () => {
    const wrapper = mount({
      render () {
        return <NLoader size="lg" show={true}/>;
      }
    });
    expect(wrapper.find('.lg-text').exists()).toBe(true);
  });
  it('render chinese characters correctly', () => {
    const wrapper = mount({
      render () {
        return <NLoader show={true} loaddingText="加载中"/>;
      }
    });
    expect(wrapper.text()).toBe('加载中');
  });
});
