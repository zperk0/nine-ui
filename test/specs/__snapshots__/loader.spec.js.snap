// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NLoader renders correctly 1`] = `<span class="loadding"><div style="height: 36px; width: 36px;"></div><span class="loadding-text md-text"></span></span>`;
